lib/internal/url.js

```diff
@@ -814,7 +831,34 @@ class URL {
     return `${constructor.name} ${inspect(obj, opts)}`;
   }
 
-  #updateContext(href) {
+  #getSearchFromContext() {
+    if (!this.#context.hasSearch) return '';
+    let endsAt = this.#context.href.length;
+    if (this.#context.hasHash) endsAt = this.#context.hash_start;
+    if (endsAt - this.#context.search_start <= 1) return '';
+    return StringPrototypeSlice(this.#context.href, this.#context.search_start, endsAt);
+  }
+
+  #getSearchFromParams() {
+    if (!this.#searchParams) return '';
+    if (!this.#searchParams.size) return '';
+    return `?${this.#searchParams.toString()}`;
```

Comment:
Not sure if it makes much change in practice, but we can skip the explicit `toString` call
```suggestion
    return `?${this.#searchParams}`;
```